src/tools/rbd/action/Kernel.cc

```diff
@@ -153,6 +153,39 @@ static int do_kernel_showmapped(Formatter *f)
   return r;
 }
 
+static void rbd_check_unsup_features(const char *poolname, const char *imgname,
+				     const char *snapname, int maperrno)
+{
+  int r;
+  uint8_t oldformat;
+  librados::Rados rados;
+  librados::IoCtx ioctx;
+  librbd::Image image;
+
+  /*
+   * kernel returns ENXIO when mapping a V2 image due to unsupported feature
+   * set or due to an invalid header for a V1 image format.
+   */
+  if (maperrno != ENXIO)
```

Comment:
It might be nice to ensure that we have helpful coverage for the most common possible failure conditions that would require checking dmesg:

1) CRUSH map incompatibilities (http://docs.ceph.com/docs/master/rados/operations/crush-map/#which-client-versions-support-crush-tunables)
2) Authentication issues
3) RBD feature incompatibilities
4) RBD "fancy" striping (RBD_FEATURE_STRIPING_V2 w/ a stripe unit != 1 and stripe width != object size)